app/com/arpnetworking/rollups/RollupGenerator.java

```diff
@@ -376,13 +376,16 @@ private LastDataPointsMessage buildLastDataPointResponse(
             final ImmutableMultimap<String, String> tags,
             final MetricsQueryResponse response,
             @Nullable final Throwable failure) {
-        final LastDataPointsMessage.Builder builder = new LastDataPointsMessage.Builder()
-                .setSourceMetricName(sourceMetricName)
-                .setRollupMetricName(rollupMetricName)
-                .setPeriod(period)
-                .setTags(tags);
+        LastDataPointsMessage result = ThreadLocalBuilder.build(
```

Comment:
Could you lift this builder into a parameter?
Then at the callsite for `buildLastDataPointResponse` you can use `ThreadLocalBuilder.build` like so:
```java
ThreadLocalBuilder.build(LastDataPointMessage.Builder.class, builder ->
    buildLastDataPointResponse(
        builder,
        sourceMetricName,
        rollupMetricName,
        period,
        tags,
        response,
        failure)
)
```
and none of the method body needs to change other than the builder instantiation. It makes me sad to see the `Optional` chaining turned into `isPresent` 😢